### DISCUSSION TITLE
Any interest in disabling "Unix timestamp" parsing for dates/time/datetimes?

### DISCUSSION THREAD

**Discussion by TomasPuverle** [pydantic/pydantic#3268] [Feature Request]
Right now, date(time) parsing supports input from an int/float unix timestamp, and I don't see a way to disable that - I am only interested in validating the "string" representations.  For example, I want to avoid a scenario where a user inadvertently specifies "2021", thinking they would get a valid date corresponding to e.g. "2021-01-01".

Would this be a useful feature?

**Comment by ojii** [comment]
We're running into the same issue. We use pydantic with fastapi and accept datetimes in various endpoints, but only want to allow ISO8601 strings (with explicit timezone), not timestamps. This causes issues when someone provides values like "20250220" (which is an invalid format but they meant February 20th 2025) which silently gets converted to 9am August 23rd 1970. For now we use a custom type that is stricter (and are still on Pydantic v1 but I tested if this is situation is improved in v2 to maybe justify an upgrade, but the behavior appears to be the same). To me this seems to be a pretty bad footgun and some builtin support for stricter handling of date(time)s might be a good idea.
